packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -259,14 +258,40 @@ export class DocumentDeltaConnection
         return this.details.initialClients;
     }
 
+    protected emitMessages(type: string, messages: IDocumentMessage[][]) {
+        // Although the implementation here disconnects the socket and does not reuse it, other subclasses
+        // (e.g. OdspDocumentDeltaConnection) may reuse the socket.  In these cases, we need to avoid emitting
+        // on the still-live socket.
+        if (!this.disposed) {
+            this.socket.emit(type, this.clientId, messages);
+        }
+    }
+
+    private static get disabledBatchManagerFeatureGate() {
+        try {
+            return localStorage !== undefined
+                && typeof localStorage === "object"
+                && localStorage.getItem(batchManagerDisabledKey) === "1";
+        } catch (e) { }
+        return false;
+    }
+
+    protected submitCore(type: string, messages: IDocumentMessage[]) {
+        if (this.isBatchManagerDisabled) {
+            this.emitMessages(type, [messages]);
+        } else {
+            this.submitManager.add(type, messages);
+        }
+    }
+
     /**
      * Submits a new delta operation to the server
      *
      * @param message - delta operation to submit
      */
     public submit(messages: IDocumentMessage[]): void {
         this.checkNotClosed();
```

Comment:
Parent checks for not closed, so I think all the methods in this class can drop checking it.
And I think parent class should have same semantics as this class, i.e. if batch manager is disabled, then it should assert if any submission is attempted while closed. The reason it was not that type of check was because of batching / delayed processing of ops (with batch manager in place), which introduced race condition, i.e. it was normal for draining to occur after closure. With synchnornos flow it's a violation of contract.

Reply:
> Parent checks for not closed

Can you clarify which parent are you referring to?